app/src/main/java/com/example/tasky/feature_authentication/domain/validation/UserDataValidatorImpl.kt

```diff
@@ -0,0 +1,77 @@
+package com.example.tasky.feature_authentication.domain.validation
+
+class UserDataValidatorImpl(
+    private val emailPatternValidator: EmailPatternValidator
+) : UserDataValidator {
```

Comment:
This now doesn't really require an abstraction anymore. I can't think of a scenario where you'd want to replace an instance of this class 😄